utils/get-git-status.js

```diff
@@ -0,0 +1,16 @@
+const execute = require('./execute')
+const assert = require('assert')
+
+async function getGitStatus () {
+  const lines = (await execute('git status')).stdout.split(/\r?\n/)
+
+  const match = lines[0].match(/^On branch (\S+)$/)
+  assert.ok(match, 'Unable to determine current branch')
+  const currentBranch = match[1]
+
+  const cleanWorkingTree = lines.includes('nothing to commit, working tree clean')
```

Comment:
Is there any better way to check if working directory is clean? Depending on wording from the output seems very unstable. E.g. for my env it's working **directory**, not working **tree**:

![](https://i.imgur.com/PBm3TYw.png)

Reply:
Should probably be using something like this https://github.com/nodegit/nodegit

Reply:
Though i do like the approach of simply executing shell commands.. They are easily repeatable manually. Tt's good to know the script is doing the *exact* same thing we are doing.

What if we just checked for "nothing to commit,"?

Reply:
> What if we just checked for "nothing to commit,"?

Fine by me :+1:

Reply:
@limonte Done in 5cf385dc60655f36f29bb4e02e5aa4adba6b58bb